jaxopt/_src/test_util.py

```diff
@@ -155,3 +155,22 @@ def lsq_linear_cube_osp(X, y, l, tol=1e-10, max_iter=None):
 def lsq_linear_cube_osp_jac(X, y, l, eps=1e-5, tol=1e-10, max_iter=None):
   return (lsq_linear_cube_osp(X, y, l + eps, tol, max_iter) -
           lsq_linear_cube_osp(X, y, l - eps, tol, max_iter)) / (2 * eps)
+
+
+def check_states_have_same_types(state1, state2):
+  if len(state1._fields) != len(state2._fields):
+    raise ValueError("state1 and state2 should have the same number of "
+                     "attributes.")
+
+  for attr1, attr2 in zip(state1._fields, state2._fields):
+    if attr1 != attr2:
+      raise ValueError("Attribute names do not agree: %s and %s." % (attr1,
+                                                                     attr2))
+
+    value1_cls_name = getattr(state1, attr1).__class__.__name__
+    value2_cls_name = getattr(state2, attr2).__class__.__name__
```

Comment:
Naive question: why not use `type` here?

Reply:
No big reason. Following your comment, I switched to type.